docs/Datalayer_for_GTM.md

```diff
@@ -0,0 +1,109 @@
+   */
+  public static function getSubscribedEvents() {
+    return [
+      DatalayerAlterEvent::EVENT_NAME => 'onDatalayerAlter',
+    ];
+  }
+
+  /**
+   * Adds category information to the datalayer.
+   */
+  public function onDatalayerAlter(DatalayerAlterEvent $event): void {
+    $node = \Drupal::routeMatch()->getParameter('node');
+    $event->datalayer['hasBenefitCategory'] = FALSE;
+
+    // make any changes need to $event->datalayer array
+    $event->datalayer['foo'] = 'bar';
+  }
+
+}
+```
+
+## Twig Templates
+
+The `html.html.twig` template outputs the payload as JSON. There are a number of flags required to render properly, the line should look like the line below. I've wrapped the line here for readability and added comments
+
+```php
+{{ datalayer |
+    // don't render accented characters as unicode entities
+    json_encode(constant('JSON_UNESCAPED_UNICODE')
+        // don't escape slashes in taxonomy URL and elsewhere
+        b-or constant('JSON_UNESCAPED_SLASHES')
+        b-or constant('JSON_PRETTY_PRINT'))
+    | raw }}
+```
+
+## Testing Changes
+
+You can use `scripts/drush/veryify-pagelayer-data.php` to validate that your changes don't change the datalayer output. This is more useful for refactoring the event subscribers or ensuring that additions to the data don't introduce unintended changes. Using the script takes some preparation:
+
+1. *Create Reference Output* The script compares Drupal outputs against a "known-good" data set, the published page CSV file created by a tome export. To prepare this file, you should export your local site using the code from the `dev` branch.
+2. *Add to the datalayer* In a new branch, create an event listener to make your changes.
+3. *Run verify script* use `drush php:script scripts/drush/verify-pagelayer-data.php`. It will fetch pages from your local site and compare the output of your datalayer to the expected values in the CSV file.
+
+The verify script takes two optional options, specified after `--`. Use `--base` if your local hostname is not reachable via `http://localhost/`. Ise the `--sample` flag for a quicker run that doesn't check every URL in the CSV file.
+
+```sh
+drush php:script scripts/drush/verify-pagelayer-data.php \
+  -- --base https://example.com/ \
+  --sample
+```
+## Known Issues and Concerns
+
+### Taxonomy Data
+
+The taxonomy URLs are generated via a system block for breadcrumbs. The output there is further processed in the theme at `usagov/templates/block--usagov-taxonomy-links.html.twig`. That template renders the taxonomy data as JSON which used to be output in the html template. Now, we still use that block to get the desired taxonomy data. However, the `usa_twig_vars` module parses the JSON from the block and adds it to the datalayer object it builds. Ideally, we should be able to get the breadcrumb info directly without having to render a block. We could do this via a separate event subscriber or as part of the initial datalayer build
```

Comment:
I would be interested in your thoughts on how else we could get the breadcrumb info! Not for this PR, of course.